packages/scene-composer/src/components/panels/AddOrRemoveOverlayButton.tsx

```diff
@@ -0,0 +1,74 @@
+import * as THREE from 'three';
+import React, { useCallback, useContext, useMemo } from 'react';
+import { useIntl } from 'react-intl';
+import { Box, Button } from '@awsui/components-react';
+
+import { COMPOSER_FEATURES, KnownComponentType } from '../../interfaces';
+import { IDataOverlayComponentInternal, useStore } from '../../store';
+import { sceneComposerIdContext } from '../../common/sceneComposerIdContext';
+import { findComponentByType } from '../../utils/nodeUtils';
+import { getGlobalSettings } from '../../common/GlobalSettings';
+import { Component } from '../../models/SceneModels';
+
+export const AddOrRemoveOverlayButton: React.FC = () => {
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const selectedSceneNodeRef = useStore(sceneComposerId)((state) => state.selectedSceneNodeRef);
+  const getSceneNodeByRef = useStore(sceneComposerId)((state) => state.getSceneNodeByRef);
+  const addComponentInternal = useStore(sceneComposerId)((state) => state.addComponentInternal);
+  const removeComponent = useStore(sceneComposerId)((state) => state.removeComponent);
+  const overlayEnabled = getGlobalSettings().featureConfig[COMPOSER_FEATURES.Overlay];
+  const selectedSceneNode = getSceneNodeByRef(selectedSceneNodeRef);
+  const intl = useIntl();
+
+  const isTagComponent = useMemo(
+    () => !!findComponentByType(selectedSceneNode, KnownComponentType.Tag),
+    [selectedSceneNode],
+  );
+
+  const isOverlayComponent = useMemo(
+    () => !!findComponentByType(selectedSceneNode, KnownComponentType.DataOverlay),
+    [selectedSceneNode],
+  );
+
+  const onClick = useCallback(() => {
+    if (!selectedSceneNodeRef) return;
+
+    if (isOverlayComponent) {
+      removeComponent(
+        selectedSceneNodeRef,
+        findComponentByType(selectedSceneNode, KnownComponentType.DataOverlay)!.ref,
+      );
+      return;
+    }
+
+    const component: IDataOverlayComponentInternal = {
+      ref: THREE.MathUtils.generateUUID(),
+      type: KnownComponentType.DataOverlay,
+      subType: Component.DataOverlaySubType.OverlayPanel,
+      valueDataBindings: [],
+      dataRows: [
+        {
+          rowType: Component.DataOverlayRowType.Markdown,
+          content: '',
+        },
+      ],
+    };
+
+    addComponentInternal(selectedSceneNodeRef, component);
+  }, [selectedSceneNodeRef, isOverlayComponent, selectedSceneNode]);
+
+  return (
+    (overlayEnabled && isTagComponent && (
+      <div style={{ overflow: 'auto' }}>
+        <Box margin={{ top: 'xs' }} float={isOverlayComponent ? 'right' : 'left'}>
+          <Button onClick={onClick}>
+            {isOverlayComponent
+              ? intl.formatMessage({ defaultMessage: 'Remove overlay', description: 'button label' })
+              : intl.formatMessage({ defaultMessage: 'Add overlay', description: 'button label' })}
+          </Button>
+        </Box>
+      </div>
+    )) ||
```

Comment:
Suggest modifying this to consistently return JSX value. JavaScript is unique in its ability to handle different types, but just as we are now trying to reduce the use of `any` we should remain consistent on return values. Example: 
`  return (

    (overlayEnabled && isTagComponent 
      ? (
        <div style={{ overflow: 'auto' }}>
          <Box margin={{ top: 'xs' }} float={isOverlayComponent ? 'right' : 'left'}>
            <Button onClick={onClick}>
              {isOverlayComponent
                ? intl.formatMessage({ defaultMessage: 'Remove overlay', description: 'button label' })
                : intl.formatMessage({ defaultMessage: 'Add overlay', description: 'button label' })}
            </Button>
          </Box>
        </div>
      )
    : (<></>)
    ));`